dotnet/src/SemanticKernel/Configuration/KernelConfigExtensions.cs

```diff
@@ -0,0 +1,199 @@
+
+namespace Microsoft.SemanticKernel.Configuration;
+
+public static class KernelConfigExtensions
+{
+    /// <summary>
+    /// Adds an Azure OpenAI text completion service to the list.
+    /// See https://learn.microsoft.com/azure/cognitive-services/openai for service details.
+    /// </summary>
+    /// <param name="config">The kernel config instance</param>
+    /// <param name="serviceId">A local identifier for the given AI service</param>
+    /// <param name="deploymentName">Azure OpenAI deployment name, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
+    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="apiVersion">Azure OpenAI API version, see https://learn.microsoft.com/azure/cognitive-services/openai/reference</param>
+    /// <param name="overwrite">Whether to overwrite an existing configuration if the same id exists</param>
+    /// <returns>Self instance</returns>
+    public static KernelConfig AddAzureOpenAITextCompletion(this KernelConfig config,
+        string serviceId, string deploymentName, string endpoint, string apiKey, string apiVersion = "2022-12-01", bool overwrite = false)
+    {
+        Verify.NotEmpty(serviceId, "The service Id provided is empty");
+
+        if (!overwrite && config.TextCompletionServices.ContainsKey(serviceId))
+        {
+            throw new KernelException(
+                KernelException.ErrorCodes.InvalidServiceConfiguration,
+                $"A text completion service with id '{serviceId}' already exists");
+        }
+
+        ITextCompletion Factory(IKernel kernel) => new AzureTextCompletion(
+            deploymentName, endpoint, apiKey, apiVersion, kernel.Log, kernel.Config.HttpHandlerFactory);
+
+        config.AddTextCompletion(serviceId, Factory, overwrite);
+
+        return config;
+    }
+
+    /// <summary>
+    /// Adds the OpenAI text completion service to the list.
+    /// See https://platform.openai.com/docs for service details.
+    /// </summary>
+    /// <param name="config">The kernel config instance</param>
+    /// <param name="serviceId">A local identifier for the given AI service</param>
+    /// <param name="modelId">OpenAI model name, see https://platform.openai.com/docs/models</param>
+    /// <param name="apiKey">OpenAI API key, see https://platform.openai.com/account/api-keys</param>
+    /// <param name="orgId">OpenAI organization id. This is usually optional unless your account belongs to multiple organizations.</param>
+    /// <param name="overwrite">Whether to overwrite an existing configuration if the same name exists</param>
+    /// <returns>Self instance</returns>
+    public static KernelConfig AddOpenAITextCompletion(this KernelConfig config,
```

Comment:
Add *Service? (i.e. AddOpenAITextCompletion**Service**)
...to match SetDefaultTextCompletionService

Reply:
lol, I actually removed "Service" it felt kind of redundant when methods end with "Completion" and "Generation". I'd prefer the shorter form, no big deal though. Thoughts?